include/task.h

```diff
@@ -32,14 +56,27 @@ struct Task {
     /* 0x18 */ u16 unk18;
 };
 
+#ifndef PORTABLE
+typedef u16 IwramNodePtr;
+typedef u32 IwramNodePtr32;
+#else
+struct IwramNode;
+
+typedef struct IwramNode *IwramNodePtr;
+typedef IwramNodePtr IwramNodePtr32;
+#endif
+
+#define IWRAM_PTR(ptr) TASK_PTR(ptr)
+
 struct IwramNode {
-    u16 next;
+    IwramNodePtr next;
     s16 state;
     u8 space[0];
 };
 
-#define TaskGetStructPtr(taskp) (void *)(IWRAM_START + (taskp)->structOffset)
-#define TaskGetParent(taskp)    (struct Task *)(IWRAM_START + (taskp)->parent)
+#define TaskGetStructPtr(taskp) (void *)TASK_PTR((taskp)->data)
+#define TaskGetParent(taskp)    (struct Task *)TASK_PTR((taskp)->parent)
+#define TaskGetNext(taskp)      (struct Task *)TASK_PTR((taskp)->next)
```

Comment:
Should this really be a macro which is available outside of the task module? I'm not sure if has anything use otherthan internally?